### ISSUE TITLE
Issue with counting number of items in a series

### ISSUE THREAD

**Issue by scare376** [Kareadita/Kavita#1697]
**Describe the bug**
Kavita is not always correctly counting the number of items in a series. It seems to be using the maxitems as total items sometimes. However this does not occur on every series.

**To Reproduce**
1. Add Count tag with total number of ch/volume
2. Scan series with kavita
3. Kavita will sometimes have an accurate count of how many items there are in total items and sometimes total items will mirror max items

**Expected behavior**
Kavita will have the correct number of total items

**Screenshots**
![opera_SGOV8RhMul](https://user-images.githubusercontent.com/2068470/207864835-78261c10-f562-4a43-a4c4-0ef61ea4ff7d.png)

**Desktop (please complete the following information):**
 - OS: Docker
 - Browser: Opera
 - Version: 0.6.1.16

**Additional context**
Let me know if you want to test with some of the files i am having the issue with.

**Comment by scare376** [comment]
Just wanted to add. I created a fresh container running on stable 0.6.1.0 and had the same results.

**Comment by majora2007** [comment]
Total Items comes from ComicInfo from the first chapter of the Series, it is not the actual number of files/chapters/volumes in the series. It maps to Count field. 

Max Items is the max chapters in the series. However, I can see that it may get skewed if there are specials. Not sure if we should ignore specials when calculating or not.

**Comment by scare376** [comment]
Not sure why this was closed. Also I don't think you understood my issue with this. Kavita has the wrong number of total items. That image shows there are only 3 items in that series but kavita is somehow reading 4 items.

**Comment by majora2007** [comment]
I closed it because the total items comes from ComicInfo tags within one of the archives. So it's just reading it from the tag. There is nothing Kavita can do. It can be fixed by updating it from within ComicInfo.

**Comment by majora2007** [comment]
Okay here is the logic:
```
TotalCount = max(all chapter.TotalCount) (TotalCount is just the Count from ComicInfo)
MaxCount = max(all chapter.count). 
if (series MaxCount != TotalCount) {
  maxVolume = max(volumes.maxnumberfromrange(volume name))
  maxChapter = max(chapters.maxnumberfromrange(chapter.number range)
  if (maxVolume === totalCount) maxCount = maxVolume;
  if (maxChapter === totalCount) maxCount =totalCount;
}

Chapter.Count comes  from:
if Number (from ComicInfo) > 0 then floor the Number
if Volume (from ComicInfo) > 0 then floor volume and take the max of Count or Volume. Count from ComicInfo
```

As you can see, all of this is just derived logic from ComicInfo. 

Can you validate `Count` for the files you showed?

**Comment by scare376** [comment]
![image](https://user-images.githubusercontent.com/2068470/225765376-6116adb4-e204-4c00-85d3-3e961e9441de.png)
This is the whats in the comicinfo for all 3 of the files in that series. I stripped them down to just this while i was testing.

**Comment by scare376** [comment]
also if MaxCount is what max items references the naming doesnt exactly make sense and the icard doesnt help to clear that up. max items sounds like it would just be the count from comicinfo while total items sounds like the number that kavita can see

**Comment by scare376** [comment]
Ok I think I found part of the issue here. The files that are affected by this dont have the number tag in the ComicInfo. Seems like the logic for getting the number of volumes might not be working for chapter.count.

**Comment by majora2007** [comment]
Let me create a unit test for this, I don't think I ran into this on my server before.

**Comment by majora2007** [comment]
Okay I'm back. I took another look at this. One thing I noticed was, like you talked about, `MaxCount` is very misleading. `MaxCount` is the max Number or Volume property in all the chapters for the Series. 

If the maxcount doesn't equal the `TotalCount` (which is the max count property), then we attempt to fallback to Volume number(from filename parsing) or Chapter number (filename parsing as well). 

We only do exact checks. If volume count equals `TotalCount` exactly, then we set them equal to each other. Same with chapter count. These 2 conditions will then `Complete` the series publication status. If they don't match exactly, we just ignore the real number of items as the ComicInfo spec doesn't imply if it's Chapter or Volumes, so we can't be sure what to set it to.
